ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HivePointLookupOptimizerRule.java

```diff
@@ -757,10 +752,15 @@ private static RexNode handleAND(RexBuilder rexBuilder, RexCall call) {
         }
       }
       // Create IN clauses
-      final List<RexNode> newOperands = createInClauses(rexBuilder, inLHSExprToRHSExprs);
+      final List<RexNode> newOperands = createInClauses(rexBuilder,
+          visitedRefs, inLHSExprToRHSExprs, inLHSExprToRHSNullableExprs);
       newOperands.addAll(operands);
       // Return node
-      return RexUtil.composeConjunction(rexBuilder, newOperands, false);
+      RexNode result = RexUtil.composeConjunction(rexBuilder, newOperands, false);
+      if (!result.getType().equals(call.getType())) {
```

Comment:
Is there any particular reason why do we need this? (I think it was not needed before)
I believe this will at most only add a cast to wieden nullability

Reply:
Once again, good catch. This compensation was coming from a previous version of the patch :(